python/pyspark/mllib/linalg.py

```diff
@@ -897,6 +914,35 @@ def __init__(self, numRows, numCols, colPtrs, rowIndices, values,
             raise ValueError("Expected rowIndices of length %d, got %d."
                              % (self.rowIndices.size, self.values.size))
 
+    def __str__(self):
+        spstr = "{0} X {1} ".format(self.numRows, self.numCols)
+        if self.isTransposed:
+            spstr += "CSRMatrix\n"
+        else:
+            spstr += "CSCMatrix\n"
+
+        for i, colPtr in enumerate(self.colPtrs[:-1]):
+            endptr = self.colPtrs[i + 1]
+            values = self.values[colPtr: endptr]
+            rowindices = self.rowIndices[colPtr: endptr]
+            for j, rowInd in enumerate(rowindices):
+                if self.isTransposed:
+                    spstr += '({0},{1}) {2}\n'.format(
+                        i, rowInd, _format_float(values[j]))
+                else:
+                    spstr += '({0},{1}) {2}\n'.format(
+                        rowInd, i, _format_float(values[j]))
+
+        return spstr[:-1]
+
+    def __repr__(self):
+        colPtrs = ", ".join([str(i) for i in self.colPtrs])
+        rowIndices = ", ".join([str(i) for i in self.rowIndices])
+        values = ", ".join([_format_float(val) for val in self.values])
+        return "SparseMatrix({0}, {1}, [{2}], [{3}], [{4}], {5})".format(
```

Comment:
This could be:
```
"SparseMatrix({0}, {1}, {2!r}, {3!r}, {4!r}, {5})".format(self.numRows, self.numCols, list(self.colPtrs), list(self.rowIndices),list(self.vlaues), self.isTransposed)
```